etl/src/main/java/org/pih/hivmigration/etl/sql/PcrTestsMigrator.groovy

```diff
@@ -0,0 +1,149 @@
+              source_encounter_date date,
+              source_infant_id int,
+              sample_id varchar(255),
+              result varchar(32), 
+              date_of_result date,              
+              comments varchar(255)            
+            );
+        ''')
+
+        setAutoIncrement('hivmigration_pcr_tests', '(select max(encounter_id)+1 from encounter)')
+
+        loadFromOracleToMySql('''
+            INSERT INTO hivmigration_pcr_tests (                
+                source_infant_id,
+                sample_id,
+                source_encounter_date,
+                result, 
+                date_of_result,
+                comments
+            ) VALUES (?, ?, ?, ?, ?, ?)
+        ''',
+                '''
+            select  INFANT_ID, 
+                    SAMPLE_ID, 
+                    case 
+                        when (SAMPLE_DATE is not null ) then to_char(SAMPLE_DATE, 'yyyy-mm-dd') 
+                        else to_char(RESULT_ENTRY_DATE, 'yyyy-mm-dd')  
+                    end as encounter_date,         
+                    lower(RESULT) as result,        
+                    case 
+                        when (DATE_RESULT_RECEIVED is not null ) then to_char(DATE_RESULT_RECEIVED, 'yyyy-mm-dd') 
+                        else to_char(RESULT_ENTRY_DATE, 'yyyy-mm-dd')  
+                    end as date_of_result,    
+                    COMMENTS
+            from HIV_PCR_TRACKING_VIEW 
+            where result is not null
+        ''')
+
+
+        executeMysql("Create Lab Results encounters",'''
+
+            SET @encounter_type_lab_results = (select encounter_type_id from encounter_type where uuid = '4d77916a-0620-11e5-a6c0-1697f925ec7b');
+            SET @form_lab_results = (select form_id from form where uuid = '4d778ef4-0620-11e5-a6c0-1697f925ec7b');
+            
+            insert into encounter(
+                   encounter_id, 
+                   uuid, 
+                   encounter_datetime, 
+                   date_created, 
+                   encounter_type, 
+                   form_id, 
+                   patient_id, 
+                   creator, 
+                   location_id) 
+            select 
+                    r.encounter_id,
+                    uuid(),
+                    r.source_encounter_date,
+                    r.source_encounter_date,
+                    @encounter_type_lab_results,
+                    @form_lab_results,
+                    i.person_id,
+                    1,
+                    1    
+            from hivmigration_pcr_tests r, hivmigration_infants i 
+            where r.source_infant_id = i.source_infant_id;                                                                                      
+        ''')
+
+        executeMysql("Create PCR results obs",'''
+
+            -- Add Date of Test Results
+            SET @date_of_test_results = (concept_from_mapping('PIH', 'DATE OF LABORATORY TEST'));
```

Comment:
Per my comment above, fine to just not create this obs for any results that don't have a "date of result received"